classes/core/DataObject.php

```diff
@@ -80,57 +70,11 @@ public function __construct()
      */
     public function getLocalizedData(string $key, ?string $preferredLocale = null, ?string &$selectedLocale = null): mixed
     {
-        foreach ($this->getLocalePrecedence($preferredLocale) as $locale) {
-            $value = & $this->getData($key, $locale);
-            if (!empty($value)) {
-                $selectedLocale = $locale;
-                return $value;
-            }
-            unset($value);
+        $value = $this->getData($key);
+        if (!is_array($value)) {
```

Comment:
@NateWr, is there a case where this happens: that there is a localized data and not an array (and we use getLocalizedData function to get that data)?
Just curious why we have added this now...

Reply:
Hi @bozana! No, not that I know of. But `LocalizedData::getBestLocalizedData()` has a type hint of `array`, so this will prevent such a case causing a fatal error.

Reply:
I am wondering if we need this part of code at all:
```
$value = $this->getData($key);
if (!is_array($value)) {
  return $value;
}
```

Reply:
I think I understand: this is only for type checking, for the function getBestLocalizedData.
Hmmm... I suppose both are OK: 
- It would be OK to get an error if the type is not an array, because then it is probably something wrong.
- But this way it is maybe also OK, because maybe it is not so important that getLocalizedData() is used correctly, for multilingual keys... 
Hmmm...